src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -118,6 +118,46 @@ SpatialPooler::SpatialPooler()
   version_ = 2;
 }
 
+SpatialPooler::SpatialPooler(vector<UInt> inputDimensions,
+                                  vector<UInt> columnDimensions,
+                                  UInt potentialRadius,
+                                  Real potentialPct,
+                                  bool globalInhibition,
+                                  Real localAreaDensity,
+                                  UInt numActiveColumnsPerInhArea,
+                                  UInt stimulusThreshold,
+                                  Real synPermInactiveDec,
+                                  Real synPermActiveInc,
+                                  Real synPermConnected,
+                                  Real minPctOverlapDutyCycles,
+                                  Real minPctActiveDutyCycles,
+                                  UInt dutyCyclePeriod,
+                                  Real maxBoost,
+                                  Int seed,
+                                  UInt spVerbosity,
+                                  bool wrapAround)
+{
+  initialize(			  inputDimensions,
```

Comment:
Call `SpatialPooler()` before `initialize` since this constructor should do basically the same thing as:

```
SpatialPooler sp;
sp.initialize(...);
```